pkg/csvparser/writer.go

```diff
@@ -59,19 +80,20 @@ func (w *Writer) Write(record []string) error {
 
 		// If we don't have to have a quoted field then just
 		// write out the field and continue to the next field.
-		if !w.fieldNeedsQuotes(field) {
+		if !w.AlwaysQuote && !w.fieldNeedsQuotes(field) {
 			if _, err := w.w.WriteString(field); err != nil {
 				return err
 			}
 			continue
 		}
 
-		if err := w.w.WriteByte('"'); err != nil {
+		if _, err := w.w.WriteRune(w.Quote); err != nil {
 			return err
 		}
+
 		for len(field) > 0 {
 			// Search for special characters.
-			i := strings.IndexAny(field, "\"\r\n")
+			i := minNonNeg(strings.IndexAny(field, "\r\n"), strings.IndexRune(field, w.Quote))
```

Comment:
Add the quote to the string before entering the loop and do a single `strings.IndexAny`. `minNonNeg` no longer needed.

Reply:
> Add the quote to the string before entering the loop and do a single `strings.IndexAny`. `minNonNeg` no longer needed.

Yes, done